xblockmufi/xblockmufi.py

```diff
@@ -129,3 +195,54 @@ def build_fragment(self,
         if fragment_js:
             fragment.initialize_js(fragment_js)
         return fragment
+
+    @XBlock.json_handler
+    def student_submit(self, submissions, suffix=''):
+        '''
+        Save student answer
+        '''
+        self.student_answer = submissions['answer']
+        return {'success':True}
+
+    @XBlock.json_handler
+    def publish_event(self, data, suffix=''):
+        try:
+            event_type = data.pop('event_type')
+        except KeyError:
+            return {'result': 'error', 'message': 'Missing event_type in JSON data'}
+
+        data['user_id'] = self.scope_ids.user_id
+        data['component_id'] = self._get_unique_id()
+        self.runtime.publish(self, event_type, data)
+
+        return {'result': 'success'}
+
+    '''
+    Util functions
+    '''
+    def load_resource(self, resource_path):
+        '''
+        Gets the content of a resource
+        '''
+        resource_content = pkg_resources.resource_string(__name__, resource_path)
+        return unicode(resource_content)
+
+    def render_template(self, template_path, context={}):
+        '''
+        Evaluate a template by resource path, applying the provided context
+        '''
+        template_str = self.load_resource(template_path)
+        return Template(template_str).render(Context(context))
+
+    def resource_string(self, path):
+        '''Handy helper for getting resources from our kit.'''
+        data = pkg_resources.resource_string(__name__, path)
+        return data.decode('utf8')
+
+    def _get_unique_id(self):
+        try:
+            unique_id = self.location.name
+        except AttributeError:
+            # workaround for xblock workbench
+            unique_id = 'workbench-workaround-id'
```

Comment:
If this is functionality missing from the workbench, it'd be nice to patch that too.